.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: ^(cmake/)
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.6.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+- repo: https://github.com/pre-commit/mirrors-clang-format
+  rev: v12.0.1
```

Comment:
We use clang 18. I assume this needs to be bumped accordingly?

Reply:
Sure, I think I read somewhere that it was v12, in GammaRay v18 is a version we skipped due a bug only fixed in v19 :) but I guess it doesn't hit you
There's also https://pre-commit.ci/ which only admins can add to the repo, it automatically commits fixes in case someone pushes without having pre-commit installed, and also poke about updates (which you don't need to merge).
I think you use restyled bot for this but `The Restyled GitHub App is deprecated and will be disabled in the coming months.` so might be a good time to also change it..

Reply:
@dantti thanks for the pointer. Yes, the `Restyled` bot is useful but sometimes also annoying because it doesn't automatically apply the required changes but only suggests via PR (or comments how to do it manually). 

Maybe @wirew0rm can have another look at this.... will merge it now as is. Thanks again for the update! :+1: